src/pages/iou/IOUTransactions.js

```diff
@@ -72,6 +104,14 @@ class IOUTransactions extends Component {
     }
 
     render() {
+        const {session, personalDetails, iouReport} = this.props;
+        const sessionEmail = lodashGet(session, 'email', '');
+        const participantEmail = sessionEmail === iouReport.managerEmail ? iouReport.ownerEmail : iouReport.managerEmail;
+
+        const currentUserName = lodashGet(personalDetails, [sessionEmail, 'firstName'], '')
+                        || Str.removeSMSDomain(sessionEmail);
+        const participantName = lodashGet(personalDetails, [participantEmail, 'firstName'], '') || Str.removeSMSDomain(participantEmail);
```

Comment:
More copy and paste logic from the `IOUAction` changes. Can we avoid this redundancy somehow?